pkg/kubectl/cmd/delete.go

```diff
@@ -233,17 +241,16 @@ func (o *DeleteOptions) Validate(cmd *cobra.Command) error {
 }
 
 func (o *DeleteOptions) RunDelete() error {
-	shortOutput := o.Output == "name"
 	// By default use a reaper to delete all related resources.
 	if o.Cascade {
-		return ReapResult(o.Result, o.f, o.Out, true, o.IgnoreNotFound, o.Timeout, o.GracePeriod, o.WaitForDeletion, shortOutput, false)
+		return o.ReapResult(o.Result, o.f, true, false)
```

Comment:
Add a TODO here, and explain why you're passing `o.Result` here.

Reply:
Get rid of factory! All this method cares about is reaper, so that's what you should put in `DeleteOptions` instead of full factory.